util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -465,7 +469,11 @@ final class AsyncStream[+A] private (private val underlying: Future[LazySeq[A]])
 }
 
 object AsyncStream {
-  import LazySeq.{Cons, Empty, One}
+  private case object Empty extends AsyncStream[Nothing]
+  private case class M[A](fas: Future[AsyncStream[A]]) extends AsyncStream[A]
```

Comment:
So if we look at the type signature `Future[AsyncStream[A]] => AsyncStream[A]`, it's really a lift (`Future[A] => AsyncStream[A]`) and then a flatten (`AsyncStream[AsyncStream[A]] => AsyncStream[A]`).

This doesn't mean I'm suggesting we call this `flatLift`.